code/game/objects/items/storage/bags.dm

```diff
@@ -351,6 +351,11 @@
 	. = ..()
 	update_icon()
 
+/obj/item/storage/bag/tray/Destroy()
+	for(var/obj/objects in contents)
+		objects.forceMove(drop_location())
+	return ..()
```

Comment:
for(var/obj/important_thing in contents)
		if(!(important_thing.resistance_flags & INDESTRUCTIBLE))
			continue
		important_thing.forceMove(drop_location())

Reply:
actually, might be worth it to have this at storage level.

Reply:
primarily since i'm pretty sure there's some hats or something that aren't /storage items that can accept small items too.